File: M.I.L.K./HomeViewController.m
//
//  HomeViewController.m
//  M.I.L.K.
//
//  Created by noah davidson on 12/12/17.
//  Copyright © 2017 noah davidson. All rights reserved.
//

#import "HomeViewController.h"
#import "ImageCollectionViewCell.h"
#import "CustomImageFlowLayout.h"
#import "DetailViewController.h"

@interface HomeViewController ()

@end

@implementation HomeViewController

#pragma mark - collectionView methods

- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath
{
    
    ImageCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"cell" forIndexPath:indexPath];
    
    NSString *imageName = indexPath.row % 2 ? @"image1" : @"image2";
    
    cell.imageView.image = [UIImage imageNamed:imageName];
    
    return cell;
}

- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
{
    return 25.0;
}

- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath
{
    [self performSegueWithIdentifier:@"openDetailView" sender:indexPath];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([segue.identifier isEqualToString:@"openDetailView"])
    {
        /*
        if (!self.customTransitioningDelegate) {
            self.customTransitioningDelegate = [ImageOpeningTransitioningDelegate new];
        }
        */
        DetailViewController *detailViewController = segue.destinationViewController;
        //detailViewController.transitioningDelegate = self.customTransitioningDelegate;
        
        if ([sender isKindOfClass:[NSIndexPath class]]) {
            NSIndexPath *indexPath = (NSIndexPath*)sender;
            
            UIImage *selectedImage = ((ImageCollectionViewCell*)[self.collectionView cellForItemAtIndexPath:indexPath]).imageView.image;
            detailViewController.imageToPresent = selectedImage;
        }
    }
}

#pragma mark - view methods
- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.collectionView.collectionViewLayout = [[CustomImageFlowLayout alloc] init];
    self.collectionView.backgroundColor = [UIColor whiteColor];
}

- (BOOL)prefersStatusBarHidden
{
    return YES;
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: M.I.L.k/HomeViewController.swift
//
//  HomeViewController.swift
//  M.I.L.k
//
//  Created by noah davidson on 12/13/17.
//  Copyright © 2017 noah davidson. All rights reserved.
//

import UIKit

/*
class HomeViewController : UIViewController, UICollectionViewDataSource, UICollectionViewDelegate{
    
    
    @IBOutlet weak var collectionView: UICollectionView!
    var collectionViewLayout: CustomImageFlowLayout!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        collectionViewLayout = CustomImageFlowLayout()
        collectionView.collectionViewLayout = collectionViewLayout
        collectionView.backgroundColor = .white
    }
    
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return 25
    }
    
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! ImageCollectionViewCell
        
        let imageName = (indexPath.row % 2 == 0) ? "image1" : "image2"
        
        cell.imageView.image = UIImage(named: imageName)
        
        return cell
    }
    
    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
        performSegue(withIdentifier: "openDetailView", sender: collectionView.cellForItem(at: indexPath))
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.identifier == "openDetailView" {
            let cell = sender as? ImageCollectionViewCell
            let detailVC = segue.destination as! DetailedImageViewController

            //THIS IS WHERE YOU DO GET REQUEST ON MESSAGES AND WHATEVER YOU NEED TO PASS FROM CELL DATA

            detailVC.selectedItem = SelectedItem(messages: [Message(name: "BOB", text:"Hello, why is this not showing up Hello, why is this not showing up Hello, why is this not showing up Hello, why is this not showing up Hello, why is this not showing up Hello, why is this not showing up Hello, why is this not showing up Hello, why is this not showing up Hello, why is this not showing up"), Message(name: "Jerry", text:"Because i said so")], image: (cell?.imageView.image)!)
        }
    }

}
 */
